### PR TITLE
Node::replace_by is not copying node properties, removed the dead code

### PR DESCRIPTION
**Pull request by nemerle**
The code to copy node properties to the new node never worked, so there is no reason to keep the useless bits in there

### CODE REVIEW

**Comment by fire** [comment]
Please show in testing how this restores the replace by usage.

**Comment by nemerle** [comment]
@fire I'm unsure what do You mean?
The original code was not copying the properties ( it was that way at least ~7 years now :smile: ), so in this PR I removed essentially dead code. 
Of course, the other way one can go with this is actually fixing the property copying, but it would likely result in unexpected behavior ( no code till today was expecting the replace_by to copy those properties )

**Comment by Calinou** [comment]
@nemerle Is there an issue on GitHub about the original behavior? If so, please link to it :slightly_smiling_face:

**Comment by nemerle** [comment]
Replace_by has a somewhat confusing parameter named `p_keep_data` and the code suggests that it once was meant to copy the properties of the 'replaced' node to a replacing one.
Currently `SceneTreeDock::replace_node` is the only place that calls it with keep_data==true and has its own property copying routine.

There are a couple of issues that are somewhat related:
#37479 ( this PR is a less impactful version of #37489 )
#3393 - replace_by is referenced here

**Comment by akien-mga** [comment]
Since this is dead code it makes sense to remove it yeah.

I'd suggest also renaming `keep_data` to `keep_groups` and updated the (currently somewhat lacking) documentation in `doc/classes/Node.xml` accordingly.

**Comment by nemerle** [comment]
Updated the PR with the new argument name + docs

**Comment by akien-mga** [comment]
Looks good. Could you squash the three commits into one? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

**Comment by nemerle** [comment]
Squashed :smile:

**Review by akien-mga — APPROVED** [review]

File: doc/classes/Node.xml [hunk 1]
```diff
@@ -623,10 +623,11 @@
 			</return>
 			<argument index="0" name="node" type="Node">
 			</argument>
-			<argument index="1" name="keep_data" type="bool" default="false">
+			<argument index="1" name="keep_groups" type="bool" default="false">
 			</argument>
 			<description>
 				Replaces a node in a scene by the given one. Subscriptions that pass through this node will be lost.
+				When [code]keep_groups[/code] is true, the [code]node[/code] will be added to the same groups that the replaced node was in.
```
Comment (pycbouh) [inline doc/classes/Node.xml:630]: > If [code]keep_groups[/code] is [code]true[/code], the [code]node[/code] is added to the same groups that the replaced node has been in.

**Review by YuriSizov — CHANGES_REQUESTED** [review]

File: doc/classes/Node.xml (hunk 1 above)
Comment (nemerle) [reply doc/classes/Node.xml]: squashed that in as well :smile:

Comment (pycbouh) [reply doc/classes/Node.xml]: I was actually suggesting a different phrasing as well as a formatting change. I think it's more correct to put it into a present tense.

Comment (nemerle) [reply doc/classes/Node.xml]: D'oh :smile:, agreed & fixed.

File: doc/classes/Node.xml [hunk 2]
```diff
@@ -623,10 +623,11 @@
 			</return>
 			<argument index="0" name="node" type="Node">
 			</argument>
-			<argument index="1" name="keep_data" type="bool" default="false">
+			<argument index="1" name="keep_groups" type="bool" default="false">
 			</argument>
 			<description>
 				Replaces a node in a scene by the given one. Subscriptions that pass through this node will be lost.
+				When [code]keep_groups[/code] is [code]true[/code], the [code]node[/code] will be added to the same groups that the replaced node is in.
```
Comment (pycbouh) [inline doc/classes/Node.xml:630]:
```suggestion
				If [code]keep_groups[/code] is [code]true[/code], the [code]node[/code] is added to the same groups that the replaced node has been in.
```

Comment (pycbouh) [reply doc/classes/Node.xml:630]: Sorry, should've used the suggestions feature from the start. I hope it's clear now what I mean with the tense.

**Review by YuriSizov — CHANGES_REQUESTED** [review]

File: doc/classes/Node.xml (hunk 2 above)
Comment (nemerle) [reply doc/classes/Node.xml:630]: Just a quick question, could 'has been in' confuse someone into thinking the 'old' node is no longer in those groups?

Comment (pycbouh) [reply doc/classes/Node.xml:630]: I just went from your original use of "was". You can use "is" if the old node is still in those groups.

Comment (nemerle) [reply doc/classes/Node.xml:630]: Ok, force push incoming :smile:

**Review by YuriSizov — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**